source/platform/file_formats/ncch.hpp

```diff
@@ -38,7 +38,37 @@ using MediaUnit32 = MediaUnit<uint32_t>;
 
 using Bytes32 = uint32_t;
 
+
 struct NCCHHeader {
+    struct TypeFlags {
+        enum ContentType {
+            Unspecified          = 0,
+            SystemUpdate         = 1,
+            InstructionManual    = 2,
+            DLPChild             = 3,
+            Trial                = 4,
+            ExtendedSystemUpdate = 5,
+        };
+        
+        enum FormType {
+            NotAssigned            = 0,
+            SimpleContent          = 1,
+            ExecutableWithoutRomFS = 2,
+            Executable             = 3,
+        };
```

Comment:
These should be `enum class`es. Might be worth moving them one level up, into NCCHHeader, if that makes sense.